packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -443,6 +444,124 @@ export class AppStudioPluginImpl {
+    }
+
+    let appDefinition: IAppDefinition;
+    if (isSPFxProject(ctx.projectSettings)) {
+      let view: any;
+      if (isLocalDebug) {
+        view = {
+          localSettings: {
+            teamsApp: {
+              teamsAppId: teamsAppId,
+            },
+          },
+        };
+      } else {
+        view = {
+          config: ctx.envInfo.config,
+          state: {
+            "fx-resource-appstudio": {
+              teamsAppId: teamsAppId,
+            },
+          },
+        };
+      }
+
+      manifestString = Mustache.render(manifestString, view);
+      manifest = JSON.parse(manifestString);
+      appDefinition = this.convertToAppDefinition(manifest, false);
+    } else {
+      const appManifest = await this.getAppDefinitionAndManifest(ctx, isLocalDebug);
+      if (appManifest.isErr()) {
+        ctx.logProvider?.error("[Teams Toolkit] Update manifest failed!");
+        const isProvisionSucceeded = !!(ctx.envInfo.state
+          .get("solution")
+          ?.get(SOLUTION_PROVISION_SUCCEEDED) as boolean);
+        if (
+          appManifest.error.name === AppStudioError.GetRemoteConfigFailedError.name &&
+          !isProvisionSucceeded
+        ) {
+          throw AppStudioResultFactory.UserError(
+            AppStudioError.GetRemoteConfigError.name,
+            AppStudioError.GetRemoteConfigError.message("Update manifest failed")
+          );
+        } else {
+          throw appManifest.error;
+        }
+      }
+      [appDefinition] = appManifest.value;
+      manifest = appManifest.value[1];
+    }
+
+    const manifestFileName =
+      `${ctx.root}/${BuildFolderName}/${AppPackageFolderName}/manifest.` +
+      (isLocalDebug ? "local" : ctx.envInfo.envName) +
+      `.json`;
+    const existingManifest = await fs.readJSON(manifestFileName);
+    delete manifest.id;
+    delete existingManifest.id;
+    if (!_.isEqual(manifest, existingManifest)) {
+      const res = await ctx.ui?.showMessage(
+        "warn",
+        "The manifest file configurations has been modified already. Do you want to continue to regenerate the manifest file and update to Teams platform?",
+        true,
+        "Preview only",
+        "Preview and update"
+      );
+
+      const error = AppStudioResultFactory.UserError(
+        AppStudioError.UpdateManifestCancelError.name,
+        AppStudioError.UpdateManifestCancelError.message(manifest.name.short)
+      );
+      if (res?.isOk() && res.value === "Preview only") {
+        this.buildTeamsAppPackage(ctx, isLocalDebug);
+        return err(error);
+      } else if (res?.isOk() && res.value === "Preview and update") {
+        this.buildTeamsAppPackage(ctx, isLocalDebug);
+      } else {
+        return err(error);
+      }
+    }
+
+    const appStudioToken = await ctx?.appStudioToken?.getAccessToken();
+    const result = await this.updateApp(
+      ctx,
+      appDefinition,
+      appStudioToken!,
+      isLocalDebug,
+      !isLocalDebug,
```

Comment:
This param is indicating if toolkit will create a new teams app.

If here is false, and the app doesn't exist(e.g. user manually delete it), will cause some 404 error.

Reply:
Yes, I'm thinking whether I should create the app for local debug if user haven't done local debug before. Maybe I can confirm with PM.

Reply:
updated